### PR TITLE
chore: Add a new sqlite based queue package

### PR DESCRIPTION
**Pull request by MohamedBassem**
chore: Add a new sqlite based queue package

### CODE REVIEW

**Review by kamtschatka — COMMENTED** [review]
code looks good, I am wondering why it is a separate db though?
I was thinking to add searching capability for all kinds of states, which is harder if we have 2 database connections.
Right now, if you go to the admin panel, you see a lot of numbers regarding bookmarks waiting for worker runs/inference runs, but that is all.
I would love to be able to click on it (or create a filter) to see all the bookmarks that are pending/had issues/whatever, so you can reschedule them.

File: packages/queue/drizzle/0000_blue_weapon_omega.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+CREATE TABLE `tasks` (
```
Comment (kamtschatka) [inline packages/queue/drizzle/0000_blue_weapon_omega.sql]: if we had the bookmark id as a field, we could do queries on them, without having to look into the payload.
To my knowledge, everything done in the worker is currently always related to a bookmark?

**Comment by MohamedBassem** [comment]
> why it is a separate db though?

Mostly for isolation purposes. I didn't want to expose hoarder to too much knowledge about what happens in the queue layer. The queues are mostly separate state that's not related to the data stored in hoarder, so I didn't see a reason to tie them to the main database. They are also kinda ephemeral. I'm planning to eventually extract it into a separate package, and having that package own the database makes things simpler.

> I would love to be able to click on it (or create a filter) to see all the bookmarks that are pending/had issues/whatever, so you can reschedule them.

The state is already reflected in the database (`bookmarks.taggingStatus` and `bookmarkLinks.crawlStatus`), so you can already do this without introspecting the queue status.

File: packages/queue/drizzle/0000_blue_weapon_omega.sql (hunk 1 above)
Comment (MohamedBassem) [reply packages/queue/drizzle/0000_blue_weapon_omega.sql]: Currently, yes. But it doesn't have to stay like that. I already have plans to include a worker to periodically reap orphaned assets for example. So I want to keep the queues generic.

Comment (kamtschatka) [reply packages/queue/drizzle/0000_blue_weapon_omega.sql]: we can have that as a nullable field then?

**Pull request merged**